### PR TITLE
Make run.py work with nerf_synthetic datasets

### PR DESCRIPTION
**Pull request by pwais**
With this change, I was able to get decent results with as few as 1000 training iters with the reference NeRF synthetic Blender scenes linked in the README

```
$ python3 scripts/run.py --scene=data/nerf_synthetic/lego/ --mode=nerf --screenshot_transforms=data/nerf_synthetic/lego/transforms_test.json --screenshot_w=800 --screenshot_h=800 --screenshot_dir=data/nerf_synthetic/lego/screenshots --save_snapshot=data/nerf_synthetic/lego/snapshot.msgpack --n_steps=1000
```

The problem is that the "NeRF Blender format" lacks the file extensions in the transforms JSON metadata.  Though some NeRF projects do either supply or allow these file extensions.  This change simply provides compatibility.  

With this change, I get outputs in `screenshots` like these:
r_0

r_1

Perhaps we could also add that training command to the root README to make it easier to try things out?

### CODE REVIEW

File: scripts/run.py [hunk 1]
```diff
@@ -256,6 +256,8 @@ def parse_args():
 				cam_matrix = f["transform_matrix"]
 				testbed.set_nerf_camera_matrix(np.matrix(cam_matrix)[:-1,:])
 				outname = os.path.join(args.screenshot_dir, os.path.basename(f["file_path"]))
+				if not outname.endswith('.png'):
```
Comment (Tom94) [inline scripts/run.py:259]: Thank you for the constribution! Could you replace this check with `if not os.path.splitext(outname)[1]:`?

Reasoning: some datasets may use `.jpg` or other extensions. We do not want to append `.png` in this case, but rather use tha datasets expected format.

Comment (pwais) [reply scripts/run.py]: good call!  yeah I guess it's fair that this code can work for more arbitrary input

**Comment by pwais** [comment]
Addressed comments.

I also believe I fixed the behavior of `screenshot_frames` to render all test transforms (in a command like `python3 scripts/run.py --scene=data/nerf_synthetic/lego/ --mode=nerf --screenshot_transforms=data/nerf_synthetic/lego/transforms_test.json --screenshot_w=800 --screenshot_h=800 --screenshot_dir=data/nerf_synthetic/lego/screenshots --save_snapshot=data/nerf_synthetic/lego/snapshot.msgpack --n_steps=1000`)  ... 

in the existing code the `if not len` doesn't work because `args.screenshot_frames` can be `None` and the `range(len(ref_transforms))` is wrong because `ref_transforms` is always a dict with two keys-- it's the 'frames' key that has all the xforms

**Comment by Tom94** [comment]
Thank you very much, looks good to me!

**Pull request merged**